crates/pipeline-manager/src/api/pipeline.rs

```diff
@@ -517,60 +505,97 @@ pub(crate) async fn post_pipeline_action(
+    context_path = "/v0",
+    security(("JSON web token (JWT) or API key" = [])),
+    params(
+        ("pipeline_name" = String, Path, description = "Unique pipeline name"),
+        ("table_name" = String, Path, description = "Unique table name"),
+        ("connector_name" = String, Path, description = "Unique input connector name"),
+        ("action" = String, Path, description = "Input connector action (one of: start, pause)")
+    ),
     responses(
-        (status = ACCEPTED
-            , description = "Request accepted."),
+        (status = OK
+            , description = "Action has been processed"),
         (status = NOT_FOUND
-            , description = "Specified pipeline id does not exist."
+            , description = "Pipeline with that name does not exist"
             , body = ErrorResponse
             , example = json!(examples::error_unknown_pipeline())),
         (status = NOT_FOUND
-                , description = "Specified endpoint does not exist."
-                , body = ErrorResponse),
-    ),
-    params(
-        ("pipeline_name" = String, Path, description = "Unique pipeline name"),
-        ("endpoint_name" = String, Path, description = "Input endpoint name"),
-        ("action" = String, Path, description = "Endpoint action [start, pause]")
+            , description = "Table with that name does not exist"
+            , body = ErrorResponse),
+        (status = NOT_FOUND
+            , description = "Input connector with that name does not exist"
+            , body = ErrorResponse),
     ),
-    context_path = "/v0",
-    security(("JSON web token (JWT) or API key" = [])),
     tag = "Pipelines"
 )]
-#[post("/pipelines/{pipeline_name}/input_endpoints/{endpoint_name}/{action}")]
-pub(crate) async fn input_endpoint_action(
+#[post("/pipelines/{pipeline_name}/tables/{table_name}/connectors/{connector_name}/{action}")]
+pub(crate) async fn post_pipeline_input_connector_action(
     state: WebData<ServerState>,
     tenant_id: ReqData<TenantId>,
-    req: HttpRequest,
+    path: web::Path<(String, String, String, String)>,
 ) -> Result<HttpResponse, ManagerError> {
-    let pipeline_name = parse_string_param(&req, "pipeline_name")?;
-    let endpoint_name = parse_string_param(&req, "endpoint_name")?;
-    let action = parse_pipeline_action(&req)?;
+    // Parse the URL path parameters
+    let (pipeline_name, table_name, connector_name, action) = path.into_inner();
 
-    state
+    // Validate action
+    if action != "start" && action != "pause" {
+        return Err(ApiError::InvalidConnectorAction { action }.into());
+    }
+
+    // Formulate and URL encode endpoint name to account for special characters.
+    // Only tables names surrounded by double quotes are case-sensitive.
+    // Case-insensitive table names are converted to lower case in the endpoints names.
+    let endpoint_name = format!(
+        "{}.{connector_name}",
+        if table_name.starts_with("\"") && table_name.ends_with("\"") {
```

Comment:
can this be shortened as `SqlIdentifier::from(table_name).name()`?

Reply:
Unfortunately not -- let's meet today to discuss how names in SQL should be referenced in the API.

Reply:
Resolved now.